src/test/java/com/redhat/cloud/event/parser/ConsoleCloudEventParserTest.java

```diff
@@ -161,6 +169,57 @@ public void shouldFailOnInvalidJson() throws IOException {
         assertThrows(RuntimeException.class, () -> consoleCloudEventParser.fromJsonString("hello world"));
     }
 
+    /**
+     * Tests that generating a Cloud Event in Java and serializing it lays a
+     * JSON that can be read back with the parser.
+     * @throws JsonProcessingException if any unexpected error occurs.
+     */
+    @Test
+    public void shouldParseFromGeneratedCloudEvent() throws JsonProcessingException {
+        final LocalDateTime ceTime = LocalDateTime.now(ZoneOffset.UTC);
+        final String ceAccountId = "ce-account-id";
+        final String ceDataSchema = "https://console.redhat.com/api/schemas/core/v1/empty.json";
+        final String ceOrgId = "ce-org-id";
+        final String ceSource = "urn:redhat:source:console:some-app";
+        final String ceSpecVersion = "1.0";
+        final String ceSubject = String.format("urn:redhat:subject:some-app:%s", UUID.randomUUID());
+        final String ceType = "console.redhat.com.some.app.type";
+        final UUID ceId = UUID.randomUUID();
+
+        final GenericConsoleCloudEvent<JsonNode> cloudEvent = new GenericConsoleCloudEvent<>();
+        cloudEvent.setTime(LocalDateTime.now(ZoneOffset.UTC));
+
+        final ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.registerModule(new JavaTimeModule());
+        objectMapper.disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS);
+
+        cloudEvent.setAccountId(ceAccountId);
+        cloudEvent.setData(objectMapper.readTree("{}"));
+        cloudEvent.setDataSchema(ceDataSchema);
+        cloudEvent.setId(ceId);
+        cloudEvent.setOrgId(ceOrgId);
+        cloudEvent.setSource(ceSource);
+        cloudEvent.setSpecVersion(ceSpecVersion);
+        cloudEvent.setSubject(ceSubject);
+        cloudEvent.setTime(ceTime);
+        cloudEvent.setType(ceType);
+
+        final String result = objectMapper.writeValueAsString(cloudEvent);
+
+        final ConsoleCloudEventParser consoleCloudEventParser = new ConsoleCloudEventParser();
+        final GenericConsoleCloudEvent<JsonNode> resultCe = consoleCloudEventParser.fromJsonString(result);
```

Comment:
This argument is throwing the following exception:

```
java.lang.IllegalArgumentException: argument "in" is null
```

Am I doing something wrong here?

Reply:
True... Let me try again. My goal is to check that `LocalDateTime`s get serialized/deserialized as expected, since I'm hitting a weird issue elsewhere, but this is getting in my way...